pkg/datastore/datastore.go

```diff
@@ -10,84 +10,140 @@ var (
 )
 
 func init() {
-	Cache = &memoryDatastore{
-		manifests: map[string]*OperatorManifest{},
+	Cache = newDataStore()
+}
+
+func newDataStore() *memoryDatastore {
+	return &memoryDatastore{
+		manifests: map[string]*operatorSourceRow{},
+		packages:  map[string]*SingleOperatorManifest{},
 		parser:    &manifestYAMLParser{},
+		walker:    &walker{},
+		bundler:   &bundler{},
 	}
 }
 
 // Reader is the interface that wraps the Read method.
 //
 // Read prepares an operator manifest for a given set of package(s)
 // uniquely represeneted by the package ID(s) specified in packageIDs. It
-// returns an instance of OperatorManifestData that has the required set of
+// returns an instance of RawOperatorManifestData that has the required set of
 // CRD(s), CSV(s) and package(s).
 //
 // The manifest returned can be used by the caller to create a ConfigMap object
 // for a catalog source (CatalogSource) in OLM.
 type Reader interface {
-	Read(packageIDs []string) (marshaled *OperatorManifestData, err error)
+	Read(packageIDs []string) (marshaled *RawOperatorManifestData, err error)
 }
 
 // Writer is an interface that is used to manage the underlying datastore
 // for operator manifest.
 type Writer interface {
-	// GetPackageIDs returns a comma separated list of IDs of
-	// all package(s) from underlying datastore.
+	// GetPackageIDs returns a comma separated list of operator ID(s). Each ID
+	// returned can be used to retrieve the manifest associated with the
+	// operator from underlying datastore.
 	GetPackageIDs() string
 
-	// Write stores the list of operator manifest(s) into datastore.s
-	Write(packages []*OperatorMetadata) error
+	// Write stores the list of operator manifest(s) into datastore.
+	Write(rawManifests []*OperatorMetadata) error
+}
+
+// operatorSourceRow is what gets stored in datastore after an OperatorSource CR
+// is reconciled.
+//
+// Every reconciled OperatorSource object has a corresponding operatorSourceRow
+// in datastore. The Writer interface accepts a raw operator manifest and
+// marshals it into this type before writing it to the underlying storage.
+type operatorSourceRow struct {
+	// RegistryMetadata uniquely identifies a given operator manifest and
+	// points to its source in remote registry.
+	RegistryMetadata RegistryMetadata
+
+	// Data is a structured representation of the given operator manifest.
+	Data StructuredOperatorManifestData
 }
 
 // memoryDatastore is an in-memory implementation of operator manifest datastore.
 // TODO: In future, it will be replaced by an indexable persistent datastore.
 type memoryDatastore struct {
-	manifests map[string]*OperatorManifest
+	manifests map[string]*operatorSourceRow
```

Comment:
Please name this variable so that it reflects a `operatorSourceRow`. Using manifests all over makes the code a bit confusing.

Reply:
Done